### PR TITLE
Add test for issue #1026

### PR DESCRIPTION
**Pull request by mroderick**
#### Purpose

This is a PR for **v1.17** branch, it adds a failing test case for #1026 
#### How to verify - mandatory
1. Check out this branch (see github instructions below)
2. `npm install`
3. `npm test`
4. Observe the magnificence of the failing test case

### CODE REVIEW

**Comment by fatso83** [comment]
Should I merge this before we have a working fix? That would leave a broken `master`.

**Comment by mroderick** [comment]
> Should I merge this before we have a working fix? That would leave a broken master.

I would say no, this is a minor issue that's been around for years.

File: test/issues/issues-test.js [hunk 1]
```diff
@@ -221,6 +221,29 @@
...
+
+            var stubbedObject = sinon.stub({
+                watch: function () {}
+            });
+
+            stubbedObject.watch();
+
+            assert.isArray(stubbedObject.watch.args);
+
+            if (oldWatch) {
```
Comment (fearphage) [inline test/issues/issues-test.js]: None of this is guaranteed to be reached it the test fails.

**Comment by mroderick** [comment]
@fatso83 @fearphage I've added a few commits to this, I believe it is complete and fixes #1026. If you agree, then I'd like to squash the `Review:` commit, and merge this.

**Comment by mroderick** [comment]
Once we merge this, we will have to port the fix to `master` also

**Comment by fatso83** [comment]
Morgan, where you supposed to squash the review commit before I merged this?

**Comment by fatso83** [comment]
Rebased in my branch:
```bash
git remote add fatso83 git@github.com:fatso83/sinon.git
git fetch fatso83
git reset --hard fatso83/add-test-for-issue-1026
```

**Comment by mroderick** [comment]
Thanks @fatso83, I've applied the rebased version from your branch

**Pull request merged**